File: src/Problems/Problem_2.hs
-- 4613732 - Completed 29.4.2013 - Learned list comprehensions.
module Problems.Problem_2
    ( problem_2
    ) where

import Utils.Sequence    

problem_2 = sum [x | x <- takeWhile (<4000000) (map fib [2..]), even x]

File: src/C_Problems/Problem_2.c
// 4613732 - Completed 20.8.2013
#include <stdlib.h>
#include <stdio.h>

int main(){
    int fib(int);
    int sum = 0;
    for (int n = 3;1;n++){
	int m = fib(n);
	if (m >= 4000000)
	    break;
	else
	    sum = (m % 2 == 0) ? sum + m : sum;
    }
    printf("%d",sum);
}

int fib(int n){
    if (n==1 || n==2)
	return 1;
    else
	return fib(n-1) + fib(n-2);
}
